### PR TITLE
Fixes #980, marker plotting 

### LINKED ISSUE #980: Markers don't seem to work properly in 0.8.x
Currently getting the error below with markers in 0.8.x. Not really sure what's broken specifically yet.

broken

### PR DESCRIPTION
**Pull request by to266**
Fixes https://github.com/hyperspy/hyperspy/issues/980

### CODE REVIEW

**Comment by dnjohnstone** [comment]
This works for me - would merge.

**Comment by francisco-dlp** [comment]
It would be nice to add the code that triggered this bug as a test.

**Comment by to266** [comment]
Not the kind of bug that you can trigger, sorry - once it's fixed, you won't see it again unless python changes (what happened in this case - `str` and `bytes` became iterable)

**Comment by francisco-dlp** [comment]
Still, a simple call to the marker function would have detected this bug, so it'll be good to have it.

**Comment by to266** [comment]
oh, I misunderstood, sorry. Will add that soon

**Comment by to266** [comment]
added

File: hyperspy/tests/drawing/test_marker.py [hunk 1]
```diff
@@ -27,17 +27,37 @@ class Test_markers:
...
+                                 y2=1.5)
         m.axes_manager = s.axes_manager
         nose.tools.assert_equal(m.get_data_position('x1'), 0)
         nose.tools.assert_equal(m.get_data_position('y1'), 1.3)
         s.axes_manager[0].index = 2
         nose.tools.assert_equal(m.get_data_position('x1'), 2)
         nose.tools.assert_equal(m.get_data_position('y1'), 1.3)
 
+    def test_iterate_strings(self):
+        s = Image(np.zeros([3, 2, 2]))
```
Comment (francisco-dlp) [inline hyperspy/tests/drawing/test_marker.py]: We shouldn't be using `Image` anymore, `Signal2D`?

File: hyperspy/tests/drawing/test_marker.py [hunk 2]
```diff
@@ -27,17 +27,37 @@ class Test_markers:
...
+        m = markers.text(x=list(range(3)),
+                         y=list(range(3)),
+                         text=['one', 'two', 'three'])
+        m.axes_manager = s.axes_manager
+        nose.tools.assert_equal(m.get_data_position('text'), 'one')
+        s.axes_manager[0].index = 2
+        nose.tools.assert_equal(m.get_data_position('text'), 'three')
+
+    def test_get_one_string(self):
+        s = Image(np.zeros([3, 2, 2]))
```
Comment (francisco-dlp) [inline hyperspy/tests/drawing/test_marker.py]: Same here.

File: hyperspy/tests/drawing/test_marker.py [hunk 3]
```diff
@@ -27,17 +27,37 @@ class Test_markers:
...
+        m = markers.text(x=list(range(3)),
+                         y=list(range(3)),
+                         text='one')
+        m.axes_manager = s.axes_manager
+        nose.tools.assert_equal(m.get_data_position('text'), 'one')
+        s.axes_manager[0].index = 2
+        nose.tools.assert_equal(m.get_data_position('text'), 'one')
+
     def test_get_data_array(self):
         s = Image(np.zeros([2, 2, 2, 2]))
```
Comment (francisco-dlp) [inline hyperspy/tests/drawing/test_marker.py]: And here.

**Comment by to266** [comment]
I thought most of these should have been done here https://github.com/hyperspy/hyperspy/pull/963 ....

**Comment by to266** [comment]
I'll create a separate PR for these

**Pull request merged**